### PR TITLE
feat(node/rollup-compat): Be compatable with rollup's `NormalizedInputOptions#input`

### PR DESCRIPTION
**Pull request by hyf0**

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2894** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2894?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2894?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2892** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2892?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2890** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2890?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/rolldown_binding/src/types/binding_normalized_input_options.rs [hunk 1]
```diff
@@ -16,4 +19,30 @@ impl BindingNormalizedInputOptions {
   pub fn shim_missing_exports(&self) -> bool {
     self.inner.shim_missing_exports
   }
+
+  // Notice: rust's HashMap doesn't guarantee the order of keys, so not sure if it's a good idea to expose it to JS directly.
+  #[napi(getter)]
```
Comment (IWANABETHATGUY) [inline crates/rolldown_binding/src/types/binding_normalized_input_options.rs:24]: Maybe we could return `Vec<(String, String)>` and convert to `{}` in js side? IIRC, Js map could preserve key order, also I assume this method is not frequently called in the real world.

Comment (hyf0) [reply crates/rolldown_binding/src/types/binding_normalized_input_options.rs:24]: Yes. Let me do it in another PR. I just didn't want to write the normalization code in js side form that time.

**Review by IWANABETHATGUY — APPROVED** [review]

**Comment by hyf0** [comment]
### Merge activity

* **Nov 24, 6:26 AM EST**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2894) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**